src/ansys/aedt/core/examples/downloads.py

```diff
@@ -0,0 +1,808 @@
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""Download example datasets from https://github.com/ansys/example-data"""
+
+from pathlib import Path
+import shutil
+import tempfile
+from typing import Callable
+from typing import Optional
+from typing import Tuple
+from typing import Union
+from urllib.parse import urljoin
+from urllib.parse import urlparse
+import urllib.request
+import zipfile
+
+from ansys.aedt.core.aedt_logger import pyaedt_logger
+from ansys.aedt.core.generic.general_methods import pyaedt_function_handler
+from ansys.aedt.core.generic.settings import settings
+from ansys.aedt.core.internal.errors import AEDTRuntimeError
+
+EXAMPLES_DATA_REPO = "https://github.com/ansys/example-data/raw/main"
+EXAMPLES_PATH = Path(tempfile.gettempdir()) / "PyAEDTExamples"
+
+
+def delete_downloads():
+    """Delete all downloaded examples to free space or update the files."""
+    shutil.rmtree(EXAMPLES_PATH, ignore_errors=True)
+
+
+def _build_safe_url(github_relative_path: str) -> str:
+    """Safely construct a URL using the user-provided input.
+
+    This function ensures that the user-provided path does not contain an unsupported scheme
+    (like 'file://', 'ftp://', or a full URL), preventing the risk of overriding the base
+    or accessing unintended resources.
+
+    Parameters
+    ----------
+    github_relative_path : str
+        A relative path provided by the user, such as ``"pyaedt/sbr/Cassegrain.aedt"``.
+
+    Returns
+    -------
+    str
+        The safely constructed URL combining the hardcoded base and the user path.
+    """
+    # Strip dangerous schemes
+    parsed = urlparse(github_relative_path)
+    if parsed.scheme:
+        raise ValueError(f"User path contains a scheme: {parsed.scheme}")
+
+    url = urljoin(EXAMPLES_DATA_REPO + "/", github_relative_path + "/")
+
+    return url
+
+
+def _download_file(
+    github_relative_path: str,
+    local_path: Optional[Union[str, Path]] = None,
+    strip_prefix: Optional[Union[str, Path]] = None,
+) -> Path:
+    """Download a file from a URL."""
+    url = _build_safe_url(github_relative_path)
+    relative_path: Path = Path(github_relative_path.strip("/"))
+
+    if strip_prefix:
+        relative_path = relative_path.relative_to(Path(strip_prefix))
+
+    if not local_path:
+        local_path = EXAMPLES_PATH / relative_path
+    else:
+        local_path = Path(local_path) / relative_path
+    local_path.parent.mkdir(parents=True, exist_ok=True)
+
+    try:
+        if not local_path.exists():
+            pyaedt_logger.debug(f"Downloading file from URL {url}")
+            urllib.request.urlretrieve(url, local_path)  # nosec
```

Comment:
If possible, I would avoid using `urllib.request.urlretrieve` directly, as it lacks proper security checks.
Instead could you use `urllib.request.urlopen` with proper validation.

But it might be overkilled in this case as we already now the origin of the file to download.